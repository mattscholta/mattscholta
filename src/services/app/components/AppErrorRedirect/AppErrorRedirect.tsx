import * as React from 'react';

interface AppErrorRedirectProps {
  className?: string;
  message?: string;
}

/**
 * @name AppErrorRedirect
 * @description This guy is used in our fetch logic
 */
const AppErrorRedirect: React.FC<AppErrorRedirectProps> = (props) => {
  const { message } = props;

  // ðŸ”Œ Short Circuit
  if (!message) return null;

  return <div className={props.className}>{message}</div>;
};

export { AppErrorRedirect };
export type { AppErrorRedirectProps };
