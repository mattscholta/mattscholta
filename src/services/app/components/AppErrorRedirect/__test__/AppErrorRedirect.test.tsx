import * as React from 'react';
import { render, RenderResult } from '@testing-library/react';
import { RecoilRoot } from 'recoil';

import { AppErrorRedirect, AppErrorRedirectProps } from '../AppErrorRedirect';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      asPath: '',
      pathname: '',
      push: jest.fn(),
      query: {},
      route: '/'
    };
  }
}));

describe('AppErrorRedirect', () => {
  const message = 'Sample error message';

  let props: AppErrorRedirectProps;
  let wrapper: RenderResult;

  const createWrapper = () => (
    <RecoilRoot>
      <AppErrorRedirect {...props} />
    </RecoilRoot>
  );

  beforeAll(() => {
    props = {};
    wrapper = render(createWrapper());
  });

  it('returns null when there is no message', () => {
    expect(wrapper.container.firstChild).toBeNull();
    expect(wrapper.container.childNodes).toMatchSnapshot('null');
  });

  it('renders an error message', () => {
    props.message = message;
    wrapper.rerender(createWrapper());

    expect(wrapper.findAllByText(message));
    expect(wrapper.container.childNodes).toMatchSnapshot('rendered');
  });
});
