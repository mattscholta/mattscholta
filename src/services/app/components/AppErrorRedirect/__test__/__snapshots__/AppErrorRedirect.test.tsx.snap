// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppErrorRedirect renders an error message: rendered 1`] = `
NodeList [
  <div>
    Sample error message
  </div>,
]
`;

exports[`AppErrorRedirect returns null when there is no message: null 1`] = `NodeList []`;
